db/db_impl/db_impl.cc

```diff
@@ -5125,6 +5126,11 @@ Status DBImpl::EndTrace() {
   return s;
 }
 
+Replayer* DBImpl::NewReplayer(const std::vector<ColumnFamilyHandle*>& handles,
+                              std::unique_ptr<TraceReader>&& reader) {
+  return new ReplayerImpl(this, handles, std::move(reader));
```

Comment:
It might be risky the replayer user forget to deconstruct the instance.

Reply:
Done